### PR TITLE
Fix linking problem with tests when systemd enabled

### PR DESCRIPTION
**Pull request by jardous**
fixes https://github.com/GENIVI/dlt-daemon/issues/98

### CODE REVIEW

**Comment by clipka** [comment]
@jardous, I guess this was fixed with PR #97 , right? Then this one can be closed.

**Comment by jardous** [comment]
this one fixes the linking so the #97 can pass

**Comment by clipka** [comment]
@jardous , do you still experience the linking error? I have successfully build #PR97 without linking problems.

**Comment by clipka** [comment]
@jardous , may I ask how to access the travis ci? How do I know what fails?

**Comment by jardous** [comment]
> @jardous , may I ask how to access the travis ci? How do I know what fails?

can you access https://travis-ci.org/jardous/dlt-daemon/branches ?

**Comment by jardous** [comment]
> @jardous , do you still experience the linking error? I have successfully build #PR97 without linking problems.

According to Juha (PS author) this PS should solve some more issues

**Comment by clipka** [comment]
OK, I got it. I will check locally. Does travis ci use your private master or the GENIVI one? This I haven't seen from first look.

**Comment by jardous** [comment]
this instance is building my fork. For the GENIVI/dlt-daemon you (admin) have to enable it in the Travis CI.

**Comment by clipka** [comment]
@gunnarx: What is the official way to go here?

**Comment by jardous** [comment]
https://docs.travis-ci.com/user/tutorial/

**Comment by gunnarx** [comment]
@clipka, sorry can you clarify the question please?  Is it about the choice of CI tool, or is it about something failing in one fork of the project but not in another.

**Comment by clipka** [comment]
@gunnarx , I am a bit concerned that we have @jardous private fork that checks the travis CI. I have not been clear about that before and I missed to ask the right question.
Threrefore, I want to clarify with you how we proceed now: I think we need to put the official GENIVI/dlt-daemon to the travis CI and not a personal fork.

**Comment by gunnarx** [comment]
@clipka Oh, now I see what you are saying.  

Yes this is not so much a policy in my opinion - it is totally wrong:

The raw text from README, taken from GENIVI/dlt-daemon, master branch:
```
Build and Test status: ![build and test status](https://travis-ci.org/jardous/dlt-daemon.svg?branch=master)
```
This indeed links to a build of [jardous' fork](https://travis-ci.org/jardous/dlt-daemon), but any user seeing this in the README should expect the "build badge" to be showing the status of _this_ fork that they are looking at (and master branch unless stated).

Another detail, it's normal to link the image to the build information.  Clicking on it now only shows the image again, instead of taking us to a page like this one which shows the [build history](https://travis-ci.org/jardous/dlt-daemon).

The exact format is up to you but let me point to a clear example, which also shows how to both fetch the image, and also link it to the details page: ([README view](https://github.com/gunnarx/vsomeip/blob/master/README.md), and [raw text](https://raw.githubusercontent.com/gunnarx/vsomeip/master/README.md)

So basically, @jardous is right in that you ought to set up your own Travis build that builds the GENIVI fork, and link the README to that.    Write a new ticket for this?

**Comment by clipka** [comment]
@gunnarx , thanks for clarification. This is enough for me to go ahead and clean this up. Thanks.
